### PR TITLE
Remove GC mode transitions from mono_array_length, and replace with mono_array_length_internal.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

**Comment by jaykrell** [comment]
appdomain-serialize-exception.exe timeout, missing GC transitions?

File: mono/metadata/object-internals.h [hunk 1]
```diff
@@ -164,7 +164,13 @@ struct _MonoString {
...
 #define mono_string_length_fast(s) ((s)->length)
 
-#define mono_array_length_fast(array) ((array)->max_length)
+/**
+ * mono_array_length_internal:
+ * \param array a \c MonoArray*
+ * \returns the total number of elements in the array. This works for
+ * both vectors and multidimensional arrays.
+ */
+#define mono_array_length_internal(array) ((array)->max_length)
```
Comment (lambdageek) [inline mono/metadata/object-internals.h]: Why rename it?

Comment (jaykrell) [reply mono/metadata/object-internals.h]: Consistency and it was little used.

Comment (jaykrell) [reply mono/metadata/object-internals.h]: Consistency and it was little used, but ok either way.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
I'm wondering if we should insert safepoints in the new functions, and therefore leave fast as it was, its name and no safepoint.

**Pull request merged**